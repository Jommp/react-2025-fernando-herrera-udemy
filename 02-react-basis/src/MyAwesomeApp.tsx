import { type CSSProperties } from 'react';

const firstName = 'Chema';
const lastName = 'MP';

const favoriteGames = ['The Last Of Us', 'Gears of war 3', 'Dofus'];
const isActive = false;

const address = {
  zipCode: '77516',
  country: 'Mexico',
};

const myStyles: CSSProperties = {
  backgroundColor: '#fafafa',
  borderRadius: 20,
  padding: 10,
  marginTop: 30,
};

export const MyAwesomeApp = () => {
  return (
    <>
      <h1> {firstName} </h1>
      <h3 data-testid='last-name'> {lastName} </h3>

      <p>{favoriteGames.join(', ')}</p>
      <p>{2 + 2}</p>

      <h1>{isActive ? 'Activo' : 'No activo'}</h1>

      <p style={myStyles}>{JSON.stringify(address)}</p>
    </>
  );
};
