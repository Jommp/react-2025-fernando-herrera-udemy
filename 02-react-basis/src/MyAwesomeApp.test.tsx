import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import { MyAwesomeApp } from './MyAwesomeApp';

describe('FirstStepsApp', () => {
  test('should render firstName', () => {
    const { container } = render(<MyAwesomeApp />);
    screen.debug();

    const h1 = container.querySelector('h1');

    expect(h1?.innerHTML).toContain('Chema');
  });

  test('should render lastName', () => {
    render(<MyAwesomeApp />);
    screen.debug();

    const h3 = screen.getByTestId('last-name');

    expect(h3.innerHTML).contain('MP');
  });

  test('should match snapshot', () => {
    const { container } =  render(<MyAwesomeApp />);

    expect(container).matchSnapshot();
  })
});
