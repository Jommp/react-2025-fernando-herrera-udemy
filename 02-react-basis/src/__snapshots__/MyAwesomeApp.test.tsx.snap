// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp > should match snapshot 1`] = `
<div>
  <h1>
     
    Chema
     
  </h1>
  <h3
    data-testid="last-name"
  >
     
    MP
     
  </h3>
  <p>
    The Last Of Us, Gears of war 3, Dofus
  </p>
  <p>
    4
  </p>
  <h1>
    No activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"77516","country":"Mexico"}
  </p>
</div>
`;
